packages/ember-runtime/lib/mixins/array.js

```diff
@@ -142,17 +142,23 @@ export function arrayContentDidChange(array, startIdx, removeAmt, addAmt) {
   let cache = meta && meta.readableCache();
 
   if (cache) {
+    let length = get(array, 'length') + removeAmt - addAmt;
```

Comment:
we should likely avoid this `get` if we are not needing it.

This is getting complicated, if we go down this path we should do a helper function that can be unit tested (and more readable).